packages/flutterfire_ui/test/table_builder_test.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:fake_cloud_firestore/fake_cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutterfire_ui/src/firestore/table_builder.dart';
+
+Future<void> main() async {
+  final instance = FakeFirebaseFirestore();
+  final collection = instance.collection('persons').withConverter<Person>(
+        fromFirestore: (snapshot, options) => Person.fromMap(snapshot.data()!),
+        toFirestore: (data, option) => data.toMap(),
+      );
+
+  final bob = Person(
+    firstName: 'Bob',
+    address: Address(street: 'Awesome Road', city: 'FlutterFire City'),
+  );
+  await collection.add(bob);
+
+  testWidgets('TableBuilder WITHOUT CelBuilder is render as expected',
+      (WidgetTester tester) async {
+    // Create the widget by telling the tester to build it.
+
+    await tester.pumpWidget(_basic(collection, null));
+
+    await tester.pumpAndSettle();
+
+    // final cityFinder = find.text('FlutterFire City');
+
+    final streetFinder =
+        find.text('{street: ${bob.address.street}, city: ${bob.address.city}}');
+    final firstNameFinder = find.text(bob.firstName);
+
+    // expect(cityFinder, findsOneWidget);
+    expect(streetFinder, findsOneWidget);
+    expect(firstNameFinder, findsOneWidget);
+  });
+
+  testWidgets('TableBuilder CelBuilder is render as expected',
+      (WidgetTester tester) async {
+    // Create the widget by telling the tester to build it.
+
+    await tester.pumpWidget(
+      _basic(collection, (data, colKey) {
+        final person = Person.fromMap(data);
+
+        switch (ColumnKey.values.asNameMap()[colKey]) {
+          case ColumnKey.firstName:
+            return Text(person.firstName);
+          case ColumnKey.address:
+            return Row(
+              children: [
+                Text(person.address.street),
+                Text(person.address.city),
+              ],
+            );
+          default:
+            return Container();
+        }
+      }),
+    );
+
+    await tester.pumpAndSettle();
+
+    final cityFinder = find.text(bob.address.city);
+    final streetFinder = find.text(bob.address.street);
+    final firstNameFinder = find.text(bob.firstName);
+
+    expect(cityFinder, findsOneWidget);
+    expect(streetFinder, findsOneWidget);
+    expect(firstNameFinder, findsOneWidget);
+  });
+}
+
+Widget _basic(
+  CollectionReference<Person> col,
+  CellBuilder? builder,
+) {
+  return MaterialApp(
+    home: FirestoreDataTable(
+      query: col,
+      columnLabels: {
+        ColumnKey.firstName.name: const Text('First Name'),
+        ColumnKey.address.name: const Text('Address'),
+      },
+      cellBuilder: builder,
+    ),
+  );
+}
+
+enum ColumnKey {
+  firstName,
+  address,
+}
+
+@immutable
+class Person {
```

Comment:
What is the purpose of using withConverter and writing such a lengthy class?

Reply:
That was in prep for FirestoreDataTable to support generics, so collection would need to be of the specified type.
But since I didn't included it in this commit this is fair to remove it and add it later on.

Since I'm using the DataGenerator shortcut, that was free, but I can understand that the purpose of a test is to understand the functionality and can add burden.  Will remove unused functionalities from the class.